lib/identity_tijuana.rb

```diff
@@ -118,6 +119,26 @@ def self.fetch_updated_users(sync_id)
       },
       false
     )
+
+    # Kick off an asynchronous donations update.
+    batch_size = Settings.tijuana.pull_batch_amount
+    # The donations cutoff ensures that donations occurring after the most
+    # recent user timestamp don't get processed. This defers the import of
+    # donations linked to new members that haven't been imported yet. If the
+    # number of users in the batch is less than the batch size, then our
+    # users should now be up-to-date, so we can safely process all available
+    # donations.
+    donations_cutoff = (batch_size && updated_users.count >= batch_size) ? last_updated_at : DateTime.now
+    sync = Sync.create!(
+      external_system: 'tijuana',
+      external_system_params: {
+        pull_job: :fetch_donation_updates,
+        time_to_run: DateTime.now,
+        donations_cutoff: donations_cutoff.strftime('%Y-%m-%d %H:%M:%S.%N')
+      }.to_json,
+      sync_type: Sync::PULL_SYNC_TYPE,
+      )
+    PullExternalSystemsWorker.perform_async(sync.id)
```

Comment:
Would this be better of being split out of the `self.fetch_updated_users` and be executed separately (i.e. by adding it as a third job type to the `PULL_JOBS` array? I don't know if member records get updated when they make a transaction, so they may well vary independently?

Reply:
Oh wait, is that due to the new members not being sync'ed yet, per your comment? 😅

Reply:
Precisely 😄 I need to chain the donations import off the users import, so that I can retain some context information (the most recent user timestamp processed and the batch size). I suppose we could have both things happening inside the one method, but I thought it was nicer to be able to separately see the outcome of each import separately in the ID syncs screen.

Reply:
Will probably revisit this when we have more dependency chains coming through in the imported data set, and see whether there isn't a better way to do handle them. Propose that this is probably OK for now.